--- openssl-1.0.1m.orig/crypto/bio/b_sock.c	2015-06-06 09:03:24.852873554 +0000
+++ openssl-1.0.1m/crypto/bio/b_sock.c	2015-06-06 09:06:59.360867173 +0000
@@ -608,7 +608,8 @@ int BIO_get_accept_socket(char *host, in
         struct sockaddr_in6 sa_in6;
 # endif
     } server, client;
-    int s = INVALID_SOCKET, cs, addrlen;
+	int s=INVALID_SOCKET,cs;
+	socklen_t addrlen;
     unsigned char ip[4];
     unsigned short port;
     char *str = NULL, *e;
@@ -689,9 +690,9 @@ int BIO_get_accept_socket(char *host, in
         if ((*p_getaddrinfo.f) (h, p, &hint, &res))
             break;
 
-        addrlen = res->ai_addrlen <= sizeof(server) ?
-            res->ai_addrlen : sizeof(server);
-        memcpy(&server, res->ai_addr, addrlen);
+        addrlen = res->ai_addrlen <= (socklen_t)sizeof(server) ?
+            res->ai_addrlen : (socklen_t)sizeof(server);
+        memcpy(&server, res->ai_addr, (size_t)addrlen);
 
         (*p_freeaddrinfo.f) (res);
         goto again;
@@ -704,7 +705,7 @@ int BIO_get_accept_socket(char *host, in
     memset((char *)&server, 0, sizeof(server));
     server.sa_in.sin_family = AF_INET;
     server.sa_in.sin_port = htons(port);
-    addrlen = sizeof(server.sa_in);
+    addrlen = (socklen_t)sizeof(server.sa_in);
 
     if (h == NULL || strcmp(h, "*") == 0)
         server.sa_in.sin_addr.s_addr = INADDR_ANY;
--- openssl-1.0.1m.orig/crypto/x509v3/v3_utl.c	2015-06-06 09:03:24.904873553 +0000
+++ openssl-1.0.1m/crypto/x509v3/v3_utl.c	2015-06-06 09:07:27.040866350 +0000
@@ -405,7 +405,7 @@ char *hex_to_string(const unsigned char
     char *tmp, *q;
     const unsigned char *p;
     int i;
-    const static char hexdig[] = "0123456789ABCDEF";
+    static const char hexdig[] = "0123456789ABCDEF";
     if (!buffer || !len)
         return NULL;
     if (!(tmp = OPENSSL_malloc(len * 3 + 1))) {
