--- openssl-1.0.1m.orig/apps/openssl.c	2015-06-06 08:08:27.936971629 +0000
+++ openssl-1.0.1m/apps/openssl.c	2015-06-06 08:15:07.824959733 +0000
@@ -337,7 +337,10 @@ int main(int Argc, char *ARGV[])
     if (i == 0) {
         if (ERR_GET_REASON(ERR_peek_last_error())
             == CONF_R_NO_SUCH_FILE) {
+#if 0 /* ANDROID */
             BIO_printf(bio_err, "WARNING: can't open config file: %s\n", p);
+#endif
+
             ERR_clear_error();
             NCONF_free(config);
             config = NULL;
--- openssl-1.0.1m.orig/apps/progs.h	2015-06-06 08:08:27.936971629 +0000
+++ openssl-1.0.1m/apps/progs.h	2015-06-06 08:20:02.344950972 +0000
@@ -147,7 +147,9 @@ FUNCTION functions[] = {
     {FUNC_TYPE_GENERAL, "ocsp", ocsp_main},
 #endif
     {FUNC_TYPE_GENERAL, "prime", prime_main},
+#if 0 /* ANDROID */
     {FUNC_TYPE_GENERAL, "ts", ts_main},
+#endif
 #ifndef OPENSSL_NO_SRP
     {FUNC_TYPE_GENERAL, "srp", srp_main},
 #endif
--- openssl-1.0.1m.orig/apps/speed.c	2015-06-06 08:08:27.920971629 +0000
+++ openssl-1.0.1m/apps/speed.c	2015-06-06 08:17:20.508955786 +0000
@@ -1760,6 +1760,7 @@ int MAIN(int argc, char **argv)
         }
     }
 
+#if 0 /* ANDROID */
     if (doit[D_IGE_128_AES]) {
         for (j = 0; j < SIZE_NUM; j++) {
             print_message(names[D_IGE_128_AES], c[D_IGE_128_AES][j],
@@ -1815,6 +1816,7 @@ int MAIN(int argc, char **argv)
         CRYPTO_gcm128_release(ctx);
     }
 # endif
+# endif
 # ifndef OPENSSL_NO_CAMELLIA
     if (doit[D_CBC_128_CML]) {
         for (j = 0; j < SIZE_NUM; j++) {
--- openssl-1.0.1m.orig/crypto/ui/ui_openssl.c	2015-06-06 08:08:28.132971623 +0000
+++ openssl-1.0.1m/crypto/ui/ui_openssl.c	2015-06-06 08:18:44.248953295 +0000
@@ -195,7 +195,7 @@
  * may eventually opt to remove it's use entirely.
  */
 
-#if !defined(TERMIOS) && !defined(TERMIO) && !defined(SGTTY)
+#if !defined(TERMIOS) && !defined(TERMIO) && !defined(SGTTY) && !defined(__ANDROID__)
 
 # if defined(_LIBC)
 #  undef  TERMIOS
